### PR TITLE
Coding Language Detection Example

### PR DESCRIPTION
**Pull request by harshp8l**
### PR Overview

- [ ] This PR requires new unit tests [y/n] (make sure tests are included)
- [ ] This PR requires to update the documentation [y/n] (make sure the docs are up-to-date)
- [ ] This PR is backwards compatible [y/n]
- [ ] This PR changes the current API [y/n]

### CODE REVIEW

**Comment by lanking520** [comment]
Could you please add licence header and add reference to the original code?

File: examples/programming-lang-detection/README.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+```python
+python test_run.py
+```
+
+This tests all files in the test folder and output looks like below:
+
+```
+Final result: 7949/8016 (0.991641716567)
+clojure - Precision:0.995991983968 Recall: 0.992015968064
```
Comment (kalyc) [inline examples/programming-lang-detection/README.md]: please use a markdown table here and round values upto 2 decimal places

File: examples/programming-lang-detection/train.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+print("Loading data...")
+x, y, z = data_helper.get_input_and_labels(file_vector_size=sequence_length, max_files=n_max_files, breakup=True)
+
+# Shuffle data
+shuffle_indices = np.random.permutation(np.arange(len(y)))
+x_shuffled = x[shuffle_indices]
+y_shuffled = y[shuffle_indices]
+
+# Setting up the model
+graph_in = Input(shape=(sequence_length, number_of_quantised_characters))
```
Comment (roywei) [inline examples/programming-lang-detection/train.py]: this part can be simplified, it's using functional model API, and plug into sequential model API.
You can just write:
```
model = Sequential()
model.add(some layers)
for i in range(0, len(filter_siezes)):
    model.add(Conv1D( ... ))
...
```

refer to https://keras.io/getting-started/functional-api-guide/
and https://keras.io/getting-started/sequential-model-guide/

File: examples/programming-lang-detection/train.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+model.add(Dense(number_of_classes))
+model.add(Activation('softmax'))
+model.compile(loss='categorical_crossentropy', optimizer='adadelta', metrics=['accuracy'])
+
+# Training model
+# ==================================================
+model.fit(x_shuffled, y_shuffled, batch_size=batch_size,
+          epochs=num_epochs, validation_split=val_split, verbose=1)
+
+model.save('save_tmp.h5')
```
Comment (kalyc) [inline examples/programming-lang-detection/train.py]: please use a more readable name for saving the model such as 'language_detection_model.h5'

File: examples/programming-lang-detection/test.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import data_helper
+import keras
+import sys
+import numpy as np
+import defs
+
+print 'usage: test.py <filename>'
```
Comment (kalyc) [inline examples/programming-lang-detection/test.py]: remove

File: examples/programming-lang-detection/README.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+in an appropriate manner.
+
+The script is able to take in a file name and detect the kind of programming language within the file as well as code snippets. Across the board we have found that it
+offers >90% accuracy for languages which we currently support of the file which is provided.
+## Usage ##
+
+Run:
+
+```
+python2.7 test.py <filename>
```
Comment (kalyc) [inline examples/programming-lang-detection/README.md]: please modify instructions - with the future lib import you can run this directly - `python test.py <file>`

File: examples/programming-lang-detection/README.md [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+```
+
+You can download a pre-trained model from [here](). This model was trained on an AWS p3.16 instance and it took under 30 minutes to train.
```
Comment (kalyc) [inline examples/programming-lang-detection/README.md]: please add link to [here]

File: examples/programming-lang-detection/README.md [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+
+> load, dump, write, stream, api, manage, broker, save, process, service
+mapping, dispatch, copy, duplicate, sample, chunk, instrument
+calculate, append, repository, facade, handler, message, invoke,
+controller, locator, customer, view, model, nav, show, new, old, legacy
+
+Data are separated into training (~2000 per language) and test (~1000 per language). In the new version, in order to train for snippets, larger files are broken down and the second and last third of the file are used in the training data. 
+
+## Run the code
+You need Keras-MXNet. The code is in python 2.7.
```
Comment (kalyc) [inline examples/programming-lang-detection/README.md]: please modify

File: examples/programming-lang-detection/README.md [hunk 8]
```diff
@@ -9,13 +9,14 @@ The script is able to take in a file name and detect the kind of programming lan
...
+You will need to have Keras-MXNet. 
 
+Run:
 ```
-python2.7 test.py <filename>
+python test.py <filename>
 ```
 
-You can download a pre-trained model from [here](). This model was trained on an AWS p3.16 instance and it took under 30 minutes to train.
+You can download a pre-trained model from [here](https://s3-us-west-2.amazonaws.com/email-boy-images/LabelBotModelFiles/lang_detection_model.h5). This model was trained on an AWS p3.16 instance and it took under 30 minutes to train.
```
Comment (kalyc) [inline examples/programming-lang-detection/README.md]: who is the owner of `email-boy-images` bucket?

Comment (roywei) [reply examples/programming-lang-detection/README.md]: If you plan to have this bucket public accessible, please make sure it's white listed and does not have any legal issue.

**Comment by kalyc** [comment]
Hello @lanking520 as discussed offline closing this PR for now.
